### ISSUE TITLE
typescript v2.0.10 can't load atom-typescript Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'"

### ISSUE THREAD

**Issue by nfdevil** [TypeStrong/atom-typescript#1135]
[Enter steps to reproduce below:]

1. npm install -g typescript (this installs version 2.0.10)
2. apm install atom-typescript
3. launch atom

If I do npm install -g typescript@1.8.10, I don't get the error. Is 2.0.10 not supported (yet)? What version should I use then? I'd like to use the 2.* features of TypeScript.

**Atom Version**: 1.12.3
**Electron Version**: 1.3.6
**System**: Microsoft Windows 8.1 Enterprise
**Thrown From**: [atom-typescript](https://github.com/TypeStrong/atom-typescript) package, v10.1.12

### Stack Trace

Failed to activate the atom-typescript package

```
At Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".


EvalError: Refused to evaluate a string as JavaScript because 'unsafe-eval' is not an allowed source of script in the following Content Security Policy directive: "script-src 'self'".

    at Object.<anonymous> (C:\Users\nilsg\AppData\Roaming\npm\node_modules\typescript\lib\typescriptServices.js:61663:23)
    at Module._compile (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.value [as .js] (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\compile-cache.js:201:21)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\native-compile-cache.js:50:27)
    at Object.makeTsGlobal (C:\Users\nilsg\.atom\packages\atom-typescript\dist\typescript\makeTypeScriptGlobal.js:16:17)
    at Object.<anonymous> (C:\Users\nilsg\.atom\packages\atom-typescript\dist\main\atomts.js:4:24)
    at Module._compile (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\native-compile-cache.js:103:30)
    at Object.value [as .js] (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\compile-cache.js:201:21)
    at Module.load (module.js:458:32)
    at tryModuleLoad (module.js:417:12)
    at Function.Module._load (module.js:409:3)
    at Module.require (module.js:468:17)
    at require (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\native-compile-cache.js:50:27)
    at Package.module.exports.Package.requireMainModule (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package.js:741:27)
    at Package.module.exports.Package.activateNow (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package.js:174:16)
    at C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package.js:157:32
    at Package.module.exports.Package.measure (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package.js:92:15)
    at C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package.js:150:26
    at Package.module.exports.Package.activate (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package.js:147:34)
    at PackageManager.module.exports.PackageManager.activatePackage (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package-manager.js:548:21)
    at C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package-manager.js:529:29
    at Config.module.exports.Config.transactAsync (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\config.js:337:18)
    at PackageManager.module.exports.PackageManager.activatePackages (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package-manager.js:524:19)
    at PackageManager.module.exports.PackageManager.activate (C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\package-manager.js:506:46)
    at C:\Users\nilsg\AppData\Local\atom\app-1.12.3\resources\app.asar\src\atom-environment.js:792:28
```

### Commands

```
```

### Config

```json
{
  "core": {
    "customFileTypes": {
      "source.ini": [
        ".buckconfig",
        ".flowconfig"
      ],
      "source.json": [
        ".arcconfig",
        "BUCK.autodeps"
      ],
      "source.python": [
        "BUCK"
      ]
    },
    "disabledPackages": [
      "markdown-pdf",
      "wrap-guide",
      "omnisharp-atom",
      "linter-jscs",
      "build",
      "busy",
      "autocomplete-paths",
      "linter-scss-lint",
      "whitespace"
    ],
    "ignoredNames": [
      ".git",
      ".hg",
      ".svn",
      ".DS_Store",
      "._*",
      "Thumbs.db",
      "*.js.map"
    ],
    "packagesWithKeymapsDisabled": [
      "open-conemu-here",
      "spell-check"
    ],
    "telemetryConsent": "limited",
    "themes": [
      "one-dark-ui",
      "monokai"
    ]
  },
  "atom-typescript": {
    "typescriptServices": "C:\\Users\\nilsg\\AppData\\Roaming\\npm\\node_modules\\typescript\\lib\\typescriptServices.js"
  }
}
```

### Installed Packages

```coffee
# User
angular2-snippets-atom, v0.7.0 (inactive)
atom-beautify, v0.29.13 (inactive)
atom-electron-snippets, v2.0.3 (inactive)
atom-handlebars, v1.3.0 (inactive)
atom-html-preview, v0.1.22 (inactive)
atom-monokai, v0.10.4 (inactive)
atom-typescript, v10.1.12 (inactive)
autocomplete-json, v5.1.0 (inactive)
autocomplete-paths, v1.0.2 (inactive)
autocomplete-python, v1.8.12 (inactive)
bezier-curve-editor, v0.7.2 (inactive)
build, v0.67.0 (inactive)
busy, v0.7.0 (inactive)
color-picker, v2.2.3 (inactive)
colorful-json, v1.0.0 (inactive)
docblockr, v0.8.7 (inactive)
dracula-theme, v1.2.8 (inactive)
editorconfig, v2.0.5 (inactive)
emmet, v2.4.3 (inactive)
emmet-snippets-compatibility, v1.0.6 (inactive)
file-icons, v1.7.25 (inactive)
firewatch-syntax, v1.0.1 (inactive)
gulp-control, v0.5.0 (inactive)
highlight-selected, v0.11.2 (inactive)
imdone-atom, v2.0.17 (inactive)
jscs-fixer, v1.2.1 (inactive)
language-gitignore, v0.3.0 (inactive)
language-powershell, v4.0.0 (inactive)
linter, v1.11.18 (inactive)
linter-eslint, v8.0.0 (inactive)
linter-flake8, v2.0.2 (inactive)
linter-jscs, v4.1.0 (inactive)
linter-jshint, v3.0.1 (inactive)
linter-jsonlint, v1.3.0 (inactive)
linter-php, v1.3.1 (inactive)
linter-sass-lint, v1.7.4 (inactive)
linter-scss-lint, v3.0.4 (inactive)
linter-tslint, v0.11.1 (inactive)
local-history, v3.4.1 (inactive)
markdown-pdf, v1.5.0 (inactive)
markdown-preview-plus, v2.4.0 (inactive)
minimap, v4.25.6 (inactive)
minimap-highlight-selected, v4.4.0 (inactive)
monokai, v0.18.0 (inactive)
monokai-seti, v0.7.0 (inactive)
monokai-shade, v0.3.5 (inactive)
omnisharp-atom, v0.30.7 (inactive)
open-conemu-here, v0.5.0 (inactive)
open-recent, v5.0.0 (inactive)
php-cs-fixer, v2.5.9 (inactive)
pigments, v0.37.0 (inactive)
pretty-json, v1.6.1 (inactive)
project-manager, v3.3.3 (inactive)
python-tools, v0.6.8 (inactive)
quick-editor, v0.4.3 (inactive)
regex-railroad-diagram, v0.16.0 (inactive)
run-in-browser, v0.1.3 (inactive)
seti-ui, v1.3.2 (inactive)
sync-settings, v0.7.2 (inactive)
tabs-to-spaces, v1.0.2 (inactive)
tag, v0.4.0 (inactive)
toggle-quotes, v1.0.0 (inactive)
trailing-spaces, v0.3.2 (inactive)
tree-ignore, v0.6.2 (inactive)
tree-view-copy-relative-path, v1.0.1 (inactive)
atom-dark-syntax, v0.27.0 (inactive)
atom-dark-ui, v0.52.0 (inactive)
atom-light-syntax, v0.28.0 (inactive)
atom-light-ui, v0.45.0 (inactive)
base16-tomorrow-dark-theme, v1.3.0 (inactive)
base16-tomorrow-light-theme, v1.3.0 (inactive)
one-dark-ui, v1.6.2 (inactive)
one-light-ui, v1.6.2 (inactive)
one-dark-syntax, v1.5.0 (inactive)
one-light-syntax, v1.5.0 (inactive)
solarized-dark-syntax, v1.0.5 (inactive)
solarized-light-syntax, v1.0.5 (inactive)
about, v1.7.0 (inactive)
archive-view, v0.62.0 (inactive)
autocomplete-atom-api, v0.10.0 (inactive)
autocomplete-css, v0.13.1 (inactive)
autocomplete-html, v0.7.2 (inactive)
autocomplete-plus, v2.31.4 (inactive)
autocomplete-snippets, v1.11.0 (inactive)
autoflow, v0.27.0 (inactive)
autosave, v0.23.1 (inactive)
background-tips, v0.26.1 (inactive)
bookmarks, v0.42.0 (inactive)
bracket-matcher, v0.82.2 (inactive)
command-palette, v0.39.0 (inactive)
deprecation-cop, v0.54.1 (inactive)
dev-live-reload, v0.47.0 (inactive)
encoding-selector, v0.22.0 (inactive)
exception-reporting, v0.40.0 (inactive)
find-and-replace, v0.202.2 (inactive)
fuzzy-finder, v1.4.0 (inactive)
git-diff, v1.1.0 (inactive)
go-to-line, v0.31.0 (inactive)
grammar-selector, v0.48.2 (inactive)
image-view, v0.60.0 (inactive)
incompatible-packages, v0.26.1 (inactive)
keybinding-resolver, v0.35.0 (inactive)
line-ending-selector, v0.5.0 (inactive)
link, v0.31.2 (inactive)
markdown-preview, v0.158.8 (inactive)
metrics, v1.1.2 (inactive)
notifications, v0.65.1 (inactive)
open-on-github, v1.2.1 (inactive)
package-generator, v1.0.1 (inactive)
settings-view, v0.243.1 (inactive)
snippets, v1.0.4 (inactive)
spell-check, v0.68.4 (inactive)
status-bar, v1.4.1 (inactive)
styleguide, v0.47.2 (inactive)
symbols-view, v0.113.1 (inactive)
tabs, v0.103.0 (inactive)
timecop, v0.33.2 (inactive)
tree-view, v0.210.0 (inactive)
update-package-dependencies, v0.10.0 (inactive)
welcome, v0.35.1 (inactive)
whitespace, v0.35.0 (inactive)
wrap-guide, v0.38.2 (inactive)
language-c, v0.54.0 (inactive)
language-clojure, v0.22.1 (inactive)
language-coffee-script, v0.48.0 (inactive)
language-csharp, v0.13.0 (inactive)
language-css, v0.40.1 (inactive)
language-gfm, v0.88.0 (inactive)
language-git, v0.15.0 (inactive)
language-go, v0.43.0 (inactive)
language-html, v0.46.1 (inactive)
language-hyperlink, v0.16.1 (inactive)
language-java, v0.24.0 (inactive)
language-javascript, v0.122.0 (inactive)
language-json, v0.18.3 (inactive)
language-less, v0.29.6 (inactive)
language-make, v0.22.2 (inactive)
language-mustache, v0.13.0 (inactive)
language-objective-c, v0.15.1 (inactive)
language-perl, v0.37.0 (inactive)
language-php, v0.37.3 (inactive)
language-property-list, v0.8.0 (inactive)
language-python, v0.45.1 (inactive)
language-ruby, v0.70.2 (inactive)
language-ruby-on-rails, v0.25.1 (inactive)
language-sass, v0.57.0 (inactive)
language-shellscript, v0.23.0 (inactive)
language-source, v0.9.0 (inactive)
language-sql, v0.25.0 (inactive)
language-text, v0.7.1 (inactive)
language-todo, v0.29.1 (inactive)
language-toml, v0.18.1 (inactive)
language-xml, v0.34.12 (inactive)
language-yaml, v0.27.1 (inactive)

# Dev
No dev packages
```

**Comment by gurdilou** [comment]
Hi, I hit that one too, trying to use transpiler version 2.0.10 instead of 2.1-dev.
Any advice  ?

Thanks !

**Comment by michael-yx-wu** [comment]
i'm also seeing this issue. i'm using atom-typescript 10.1.12

**Comment by starstuffharvestingstarlight** [comment]
I had this issue with 2.0.10, solved it by installing 2.1 globally with ``sudo npm i -g typescript@^2.1``

**Comment by Goodwine** [comment]
Same as #1124 
On the other bug report it was suggested to use loophole, although I haven't tried that.
